x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/ReplaceSingleMultiMatchWithMatch.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.optimizer.rules.logical;
+
+import org.elasticsearch.xpack.esql.core.expression.Expression;
+import org.elasticsearch.xpack.esql.core.expression.Literal;
+import org.elasticsearch.xpack.esql.core.expression.MapExpression;
+import org.elasticsearch.xpack.esql.core.tree.Source;
+import org.elasticsearch.xpack.esql.expression.function.fulltext.Match;
+import org.elasticsearch.xpack.esql.expression.function.fulltext.MultiMatch;
+import org.elasticsearch.xpack.esql.plan.logical.Filter;
+import org.elasticsearch.xpack.esql.plan.logical.LogicalPlan;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.elasticsearch.xpack.esql.core.type.DataType.KEYWORD;
+
+/**
+ * Replaces {@link MultiMatch} with {@link Match}, iff {@link MultiMatch} only specifies one text field.
+ * This enables matching on a single `semantic_text` field as {@link MultiMatch} doesn't work with `semantic_text`.
+ */
+public final class ReplaceSingleMultiMatchWithMatch extends OptimizerRules.OptimizerRule<Filter> {
+    @Override
+    protected LogicalPlan rule(Filter filter) {
+        Expression condition = filter.condition();
+
+        if (condition instanceof MultiMatch multiMatch && multiMatch.fields().size() == 1) {
+            // MULTI_MATCH and MATCH have a subset of common options
+            Expression filteredOptions = filterOptionsForMatch(multiMatch.options());
```

Comment:
Will add a test making sure options are passed correctly from `MultiMatch` to `Match`